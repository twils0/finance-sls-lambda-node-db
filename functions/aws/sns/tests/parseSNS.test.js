const { parseSNS } = require('../parseSNS');

describe('functions', () => {
  describe('aws', () => {
    describe('sns', () => {
      describe('parseSNS', () => {
        it('fails and returns unexpected error, invalid sns event', async () => {
          const emptyEvent = {};

          try {
            await parseSNS(JSON.stringify(emptyEvent));
          } catch (errorCatch) {
            expect(errorCatch).toMatchSnapshot();
          }
        });

        it('correctly parses sns event', async () => {
          const message = { test: 'testMessage' };
          const event = {
            Records: [
              {
                Sns: {
                  Message: JSON.stringify(message),
                },
              },
            ],
          };

          const payload = await parseSNS(event);

          expect(payload).toEqual(message);
        });
      });
    });
  });
});
