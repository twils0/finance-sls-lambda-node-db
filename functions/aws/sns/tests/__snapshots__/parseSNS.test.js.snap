// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions aws sns parseSNS fails and returns unexpected error, invalid sns event 1`] = `"unexpected error - invalid sns event"`;
